### PR TITLE
FF-62 Implement Planned Api Interfaces

### PR DESCRIPTION
**Pull request by open-schnick**
dnm!

### CODE REVIEW

**Review by qvalentin — APPROVED** [review]
Maybe we should talk about the "update user" interface again. It would send the users password every time you just want to update your username. A seperated interface just for changing the password would be more secure.

**Comment by open-schnick** [comment]
> Maybe we should talk about the "update user" interface again. It would send the users password every time you just want to update your username. A seperated interface just for changing the password would be more secure.

Actually that depends on the implemeted logic. You can leave fields empty = null in the request, if you dont want to change them. They are not annotated as `NotNull`

File: src/main/java/de/filefighter/rest/domain/user/exceptions/UserNotAuthenticatedAdvise.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class UserNotAuthenticatedAdvise {
+
+    @ResponseBody
+    @ExceptionHandler(UserNotAuthenticatedException.class)
+    @ResponseStatus(HttpStatus.UNAUTHORIZED)
+    ResponseEntity<ServerResponse> employeeNotFoundHandler(UserNotAuthenticatedException ex) {
```
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/user/exceptions/UserNotAuthenticatedAdvise.java:19]: ups :D

File: src/main/java/de/filefighter/rest/domain/token/data/dto/RefreshToken.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import de.filefighter.rest.domain.user.data.dto.User;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class RefreshToken {
+    private final String refreshToken;
+    private final User user;
```
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/token/data/dto/RefreshToken.java:11]: we can think about that. but generally this token should work for more than one month. Default would be around one year.

File: src/main/java/de/filefighter/rest/domain/filesystem/type/FileSystemType.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package de.filefighter.rest.domain.filesystem.type;
```
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/filesystem/type/FileSystemType.java:1]: Frontend / FileHandler :D

File: src/main/java/de/filefighter/rest/domain/permission/rest/PermissionRestController.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @GetMapping(FS_BASE_URI+"{fsItemId}/permission")
+    public EntityModel<PermissionSet> getPermissionsOfFileOrFolder(
+            @PathVariable long fsItemId,
+            @RequestHeader(value = "Authorization", defaultValue = AUTHORIZATION_BEARER_PREFIX + "token") String accessToken
+    ){
+
+        LOG.info("Requested PermissionSet for FileSystemItem {}", fsItemId);
+        return permissionsRestService.updatePermissionSetByIdAndToken(fsItemId, accessToken);
```
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/permission/rest/PermissionRestController.java:31]: ups thanks.

File: src/main/java/de/filefighter/rest/domain/user/exceptions/UserAlreadyExistsException.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package de.filefighter.rest.domain.user.exceptions;
+
+public class UserAlreadyExistsException extends RuntimeException {
+
+    public UserAlreadyExistsException(Integer id) {
+        super("A employee already exists with the id " + id);
```
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/user/exceptions/UserAlreadyExistsException.java:6]: ups :DD

File: src/main/java/de/filefighter/rest/domain/token/data/dto/RefreshToken.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/token/data/dto/RefreshToken.java:11]: We can implement that later, i will add an todo.

**Pull request merged**